src/selectors_vm/stack.rs

```diff
@@ -159,16 +236,18 @@ impl<E: ElementData> Stack<E> {
     pub fn pop_up_to(
         &mut self,
         local_name: LocalName,
-        mut popped_element_data_handler: impl FnMut(E),
+        popped_element_data_handler: impl FnMut(E),
     ) {
-        for i in (0..self.items.len()).rev() {
-            if self.items[i].local_name == local_name {
-                for item in self.items.drain(i..self.items.len()) {
-                    popped_element_data_handler(item.element_data);
-                }
-
-                break;
-            }
+        let result =
```

Comment:
Nit: let's rename this variable to better reflect its meaning,  e.g. `last_popped_item_idx` or something like that.